File: src/data/courses/dsa/two-pointers-sliding-window/two-pointers-same-direction/remove-duplicates-sorted-array/code/python/remove_duplicates_sorted_array_sol2.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # Optimal: O(1) space — Slow/fast pointers
        j = 0
        for i in range(1, len(nums)):
            if nums[i] != nums[j]:
                j += 1
                nums[j] = nums[i]
        return j + 1

File: src/data/courses/dsa/two-pointers-sliding-window/two-pointers-same-direction/remove-duplicates-sorted-array/code/java/RemoveDuplicatesSortedArraySol2.java
class Solution {
    // Optimal: O(1) space — Slow/fast pointers
    public int removeDuplicates(int[] nums) {
        int j = 0;
        for (int i = 1; i < nums.length; i++)
            if (nums[i] != nums[j]) nums[++j] = nums[i];
        return j + 1;
    }
}